### PR TITLE
CookieUtils - Added more utils

### PR DESCRIPTION
**Pull request by neteinstein**

### CODE REVIEW

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt [hunk 1]
```diff
@@ -34,28 +34,32 @@ object CookiesUtils {
...
 
-    fun getCookie(url: String, cookieName: String): String? {
+    fun getCookie(url: String, cookieName: String): Map<String,String> {
         var cookieValue: String? = null
 
         val cookieManager = CookieManager.getInstance()
         val cookies = cookieManager.getCookie(url)
 
-        val cookieArray = cookies.split(";").dropLastWhile { it.isEmpty() }.toTypedArray()
+        return stringCookiesToMap(cookies).filterNot { it.key.contains(cookieName) }
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: unnecessary empty line

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt [hunk 2]
```diff
@@ -34,28 +34,32 @@ object CookiesUtils {
         return cookiesMap
     }
 
-    fun getCookie(url: String, cookieName: String): String? {
+    fun getCookie(url: String, cookieName: String): Map<String,String> {
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: missing space after comma

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt [hunk 3]
```diff
@@ -34,28 +34,32 @@ object CookiesUtils {
...
-            if (nameAndCookie.contains(cookieName)) {
+    fun stringCookiesToMap(cookies: String): HashMap<String, String> {
+        val cookiesMap = HashMap<String, String>()
 
-                val cookie =
-                    nameAndCookie.split("=", limit=2).dropLastWhile { it.isEmpty() }.toTypedArray()
+        val temp = cookies.split(";").map { it.trim() }.dropLastWhile { it.isEmpty() }
+            .toTypedArray()
+
+        for (ar1 in temp) {
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: can we have better names ?

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt (hunk 1 above)
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: Why are we filter out cookies with that `cookieName`? from the method name I'd assume the contrary.

**Review by bruno-garces — CHANGES_REQUESTED** [review]

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt (hunk 2 above)
Comment (Jmvcaseiro) [reply base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: Isn't this method just a particular case of the method above ("getCookiesMap")? From what I can see, the method above takes the cookies, parses them and returns a map with them. This method takes the cookies, parses them and then filters them. Unless there is something I'm missing, couldn't we use the method above and filter the output?

**Review by ritatferreira — APPROVED** [review]

File: base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt [hunk 4]
```diff
@@ -14,48 +14,37 @@ object CookiesUtils {
...
-                LOG.e(LOG_TAG, "Cookie is malformed, skipping...")
-            }
-        }
-        return cookiesMap
+        return stringCookiesToMap(cookies)
     }
 
-    fun getCookie(url: String, cookieName: String): String? {
+    fun getCookie(url: String, cookieName: String): Map<String, String> {
         var cookieValue: String? = null
```
Comment (Jmvcaseiro) [inline base/src/main/java/com/mindera/skeletoid/webview/cookies/CookiesUtils.kt]: can't this be removed?

**Review by bruno-garces — APPROVED** [review]

**Review by djrsousa — APPROVED** [review]

**Review by Jmvcaseiro — APPROVED** [review]

**Pull request merged**